app/src/main/java/com/example/carpool/ui/activities/MainActivity.java

```diff
@@ -17,5 +35,20 @@ protected void onCreate(Bundle savedInstanceState) {
                     .replace(R.id.fragment_container, new LoginFragment())
                     .commit();
         }
+
+
+        getOnBackPressedDispatcher().addCallback(this, new OnBackPressedCallback(true) {
+            @Override
+            public void handleOnBackPressed() {
+
+                if (getSupportFragmentManager().getBackStackEntryCount() > 0) {
+                    getSupportFragmentManager().popBackStack(null, FragmentManager.POP_BACK_STACK_INCLUSIVE);
+                }
+
+                getSupportFragmentManager().beginTransaction()
+                        .replace(R.id.fragment_container, new RideOffersFragment())
+                        .commit();
```

Comment:
```suggestion
                getSupportFragmentManager()
                        .beginTransaction()
                        .replace(R.id.fragment_container, new RideOffersFragment())
                        .commit();
```

So it's the same everywhere we use it